### PR TITLE
HOSTEDCP-827: Add root volume encryption e2e test

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-827](https://issues.redhat.com/browse/HOSTEDCP-827)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/nodepool_test.go [hunk 1]
```diff
@@ -44,14 +49,45 @@ func TestNodePool(t *testing.T) {
 	// We set replicas to 0 in order to allow the inner tests to
 	// create their own NodePools with the proper replicas
 	clusterOpts.NodePoolReplicas = 0
-	guestCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
-	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, guestCluster)
+	hostedCluster := e2eutil.CreateCluster(t, ctx, mgmtClient, &clusterOpts, globalOpts.Platform, globalOpts.ArtifactDir)
+	guestClient := e2eutil.WaitForGuestClient(t, ctx, mgmtClient, hostedCluster)
```
Comment (jparrill) [inline test/e2e/nodepool_test.go:53]: I would like to align the client and hostedCluster names, just for convention

File: test/e2e/nodepool_test.go [hunk 2]
```diff
@@ -44,14 +49,45 @@ func TestNodePool(t *testing.T) {
...
+
+	// Get the newly created defautlt NodePool
+	nodepools := &hyperv1.NodePoolList{}
+	if err := mgmtClient.List(ctx, nodepools, crclient.InNamespace(hostedCluster.Namespace)); err != nil {
+		t.Fatalf("failed to list nodepools in namespace %s: %v", hostedCluster.Namespace, err)
+	}
+	if len(nodepools.Items) != 1 {
```
Comment (jparrill) [inline test/e2e/nodepool_test.go:60]: what if the test implies HA? maybe we just should check that is not 0?

File: test/e2e/nodepool_test.go [hunk 3]
```diff
@@ -44,14 +49,45 @@ func TestNodePool(t *testing.T) {
...
+	nodePoolTests := []NodePoolTestCase{
+		{
+			name: "TestKMSRootVolumeEncryption",
+			test: NewKMSRootVolumeTest(hostedCluster, clusterOpts),
+		},
+		// reuse same test with different input
+		// {
+		// 	name: "TestKMSRootVolumeEncryption Custom NodePool",
+		// 	test: NewKMSRootVolumeTest(hostedCluster, clusterOpts),
+		// 	manifestBuilder: myCustomeNodePoolBuilder,
```
Comment (jparrill) [inline test/e2e/nodepool_test.go:76]: comment typo "Custome"

**Review by jparrill — COMMENTED** [review]
I like the way you're changing this, dropped some minor comments.

Maybe we will need to change it in the future to support HA with multiple nodepools in the same testcase, but I think it's fine for now :)

File: test/e2e/nodepool_test.go (hunk 2 above)
Comment (muraee) [reply test/e2e/nodepool_test.go:60]: AFAIK, the cli will only create 1 nodePool by default, no?

Comment (jparrill) [reply test/e2e/nodepool_test.go:60]: yep, we need to have this in mind if the HostedCluster definition changes in the future.

**Pull request merged**